lms/djangoapps/commerce/api/v1/tests/test_views.py

```diff
@@ -32,21 +33,41 @@ def setUp(self):
                                                      currency=u'USD', sku=u'ABC123')
 
     @staticmethod
-    def _serialize_course_mode(course_mode):
+    def _serialize_course_mode(course_mode, verification_deadline=None):
         """ Serialize a CourseMode to a dict. """
         # encode the datetime (if nonempty) using DRF's encoder, simplifying
         # equality assertions.
         expires = course_mode.expiration_datetime
+        json_encoder = JSONEncoder()
         if expires is not None:
-            expires = JSONEncoder().default(expires)
+            expires = json_encoder.default(expires)
+
+        if verification_deadline is not None:
+            verification_deadline = json_encoder.default(verification_deadline)
+
         return {
             u'name': course_mode.mode_slug,
             u'currency': course_mode.currency.lower(),
             u'price': course_mode.min_price,
             u'sku': course_mode.sku,
             u'expires': expires,
+            u'verification_deadline': verification_deadline,
+        }
+
+    @classmethod
+    def _serialize_course(cls, course, modes=None, verification_deadline=None):
+        """ Serializes a course to a Python dict. """
+        data = {
+            u'id': unicode(course.id),
+            u'name': unicode(course.display_name),
+            u'modes': []
         }
 
+        modes = modes or []
+        data[u'modes'] = [cls._serialize_course_mode(mode, verification_deadline) for mode in modes]
```

Comment:
this passes through the verification_deadline to all the modes, apparently without checking whether verification deadline is relevant to the mode.   would it make more sense to filter that down, the way we expect an API client to?

speaking of, the outcome of a call in which more than one mode is present and more than one mode specifies a verification deadline, and those deadlines don't match, appears to be non-deterministic.  I'm not sure whether we really need to fix that as much as document the assumption that, across all modes, at most one verification deadline will be passed.